src/odemis/driver/smaract.py

```diff
@@ -1188,9 +1188,9 @@ def __init__(self, name, role, locator, axes, ref_on_init=False, linear_speed=0.
         axes_def = {}  # axis name -> Axis object
         self._locator = locator
 
-        # Be sure that the axes are defined in the controller
-        if set(axes.keys()) < {'x', 'y', 'z', 'rx', 'rz'}:
-            raise ValueError("Some of the axes are not allowed, it should have only x, y, z, rx, rz")
+        # Require the user to define all 5 axes: x, y, z, rx, rz
+        if set(axes.keys()) == {'x', 'y', 'z', 'rx', 'rz'}:
```

Comment:
You need to invert this test. Otherwise it'll fail all the time. Did you try the test cases? If so, that is also a sign that something is wrong because they should have failed.

Reply:
I think I forgot to run the test. But now it's fixed!